Add pagination support to events based cache

faisal-memon opened the pull request:
<!--
1. If this is your first PR, please read our contributor guidelines
https://github.com/spiffe/spiffe/blob/master/CONTRIBUTING.md
https://github.com/spiffe/spire/blob/main/CONTRIBUTING.md

2. Please remember to include a DCO on every commit (`git commit -s`)
https://github.com/apps/dco
-->

**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
<!-- Please provide a description of the affected functionality -->
Events based cache

**Description of change**
<!-- Please provide a description of the change -->
The events based cache lists all entries on start up. This change paginates that initial list entries request.

~~Marking as draft until I have some unit tests for it.~~

**Which issue this PR fixes**
<!-- optional. `fixes #<issue number>` format will close an issue when this PR is merged -->
fixes #4735

azdagron reviewed (commented):
Thanks @faisal-memon! Looks good. Paging behavior should probably be covered by a unit test before we merge.

faisal-memon commented:
> Thanks @faisal-memon! Looks good. Paging behavior should probably be covered by a unit test before we merge.

@azdagron Thanks for reviewing. Added the unit tests and moved this out of draft.
